bindings/wasm/examples/worker2.test.ts

```diff
@@ -0,0 +1,103 @@
+import {WebIO} from '@gltf-transform/core';
+import assert from 'node:assert';
+import {afterEach, expect, suite, test} from 'vitest';
+
+import {readMesh, setupIO} from './gltf-io';
+// @ts-ignore
+import {examples} from './public/examples.js';
+import {cleanup, evaluateCADToModel, module} from './worker';
+
+const io = setupIO(new WebIO());
+
+async function runExample(name: string) {
+  const code = examples.functionBodies.get(name);
+  const result = await evaluateCADToModel(code);
+  module.cleanup();
+  cleanup();
+  assert.ok(result?.glbURL);
+  const docIn = await io.read(result.glbURL);
+  URL.revokeObjectURL(result.glbURL);
+  const nodes = docIn.getRoot().listNodes();
+  for (const node of nodes) {
+    const docMesh = node.getMesh();
+    if (!docMesh) {
+      continue;
+    }
+    const {mesh} = readMesh(docMesh)!;
+    const manifold = new module.Manifold(mesh as any);
+    const prop = manifold.getProperties();
+    const genus = manifold.genus();
+    manifold.delete();
+    return {...prop, genus};
+  }
+  assert.ok(false);
+}
+
+// allow vitest to report progress after each test
+// before going into heavy computation which blocks main thread
+afterEach(async () => {await new Promise(resolve => setTimeout(resolve, 500))})
+
+suite('Examples', () => {
+  test('Intro', async () => {
+    const result = await runExample('Intro');
+    expect(result.genus).to.equal(5, 'Genus');
+    expect(result.volume).to.be.closeTo(203164, 1, 'Volume');
+    expect(result.surfaceArea).to.be.closeTo(62046, 1, 'Surface Area');
+  });
+
+  test('Tetrahedron Puzzle', async () => {
+    const result = await runExample('Tetrahedron Puzzle');
+    expect(result.genus).to.equal(0, 'Genus');
+    expect(result.volume).to.be.closeTo(7240, 1, 'Volume');
+    expect(result.surfaceArea).to.be.closeTo(3235, 1, 'Surface Area');
+  });
+
+  test('Rounded Frame', async () => {
+    const result = await runExample('Rounded Frame');
+    expect(result.genus).to.equal(5, 'Genus');
+    expect(result.volume).to.be.closeTo(270807, 10, 'Volume');
+    expect(result.surfaceArea).to.be.closeTo(74599, 1, 'Surface Area');
+  });
+
+  test('Heart', async () => {
+    const result = await runExample('Heart');
+    expect(result.genus).to.equal(0, 'Genus');
+    expect(result.volume).to.be.closeTo(3.342, 0.001, 'Volume');
+    expect(result.surfaceArea).to.be.closeTo(11.51, 0.01, 'Surface Area');
+  });
+
+  test('Scallop', async () => {
+    const result = await runExample('Scallop');
+    expect(result.genus).to.equal(0, 'Genus');
+    expect(result.volume).to.be.closeTo(41400, 100, 'Volume');
+    expect(result.surfaceArea).to.be.closeTo(7770, 10, 'Surface Area');
+  });
+
+  test('Torus Knot', async () => {
+    const result = await runExample('Torus Knot');
+    expect(result.genus).to.equal(1, 'Genus');
+    expect(result.volume).to.be.closeTo(20786, 1, 'Volume');
+    expect(result.surfaceArea).to.be.closeTo(11177, 1, 'Surface Area');
+  });
```

Comment:
I'm seeing something odd here. When running this test alone, it's passing, but when running it together with `Rounded Frame`, it fails. I'm wondering if this is because I forgot to cleanup some resource in between tests. (test duration seems also affected depending on how tests are run.)


```
$ npm run test worker2.test -- -t 'Torus'
...
 ✓ worker2.test.ts (9) 6646ms
   ✓ Examples (9) 6644ms
     ...
     ✓ Torus Knot 6642ms
```

```
$ npm run test worker2.test -- -t 'Rounded|Torus'
 RUN  v1.6.0 /home/hiroshi/code/others/manifold/bindings/wasm/examples

 ❯ worker2.test.ts (9) 28180ms
   ❯ Examples (9) 28178ms
    ...
     ✓ Rounded Frame 8700ms
     × Torus Knot 19477ms


⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  worker2.test.ts > Examples > Torus Knot
AssertionError: Volume: expected 20960.73046875 to be close to 20786 +/- 1
 ❯ worker2.test.ts:79:33
     77|     const result = await runExample('Torus Knot');
     78|     expect(result.genus).to.equal(1, 'Genus');
     79|     expect(result.volume).to.be.closeTo(20786, 1, 'Volume');
       |                                 ^
     80|     expect(result.surfaceArea).to.be.closeTo(11177, 1, 'Surface Area');
     81|   });
```

I would appreciate if you can spot something I did wrong. I mostly copied `worker.test.ts` but maybe one notable difference is that previously it has a separate `Module` initialization and it's used to instantiate `Manifold`

https://github.com/elalish/manifold/blob/81454c28d5f0586b0fac060be5b67541b27d5946/bindings/wasm/examples/worker.test.js#L27-L28

https://github.com/elalish/manifold/blob/81454c28d5f0586b0fac060be5b67541b27d5946/bindings/wasm/examples/worker.test.js#L71-L73

Reply:
Good catch - I was neglecting to reset our static properties, which was probably creating a similar bug in our app. The old test wasn't catching this because it created and destroyed a worker for each test (unlike how the app works), so this new architecture is definitely better.